Newsletters: add 366 (2025-08-08) (#2448)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2025-08-08-newsletter.md

## News

- **Continued discussion about lowering the minimum relay feerate:**
  Gloria Zhao [posted][zhao minfee] to Delving Bitcoin about lowering
  the [default minimum relay feerate][topic default minimum transaction
  relay feerates] by 90% to 0.1 sat/vbyte.  She encouraged conceptual
  discussion about the idea and how it might affect other software.  For
  concerns specific to Bitcoin Core, she linked to a [pull
  request][bitcoin core #33106].

- **Peer block template sharing to mitigate problems with divergent mempool policies:**
  Anthony Towns [posted][towns tempshare] to Delving Bitcoin to suggest
  full node peers occasionally send each other their current template
  for the next block using [compact block relay][topic compact block
  relay] encoding.  The receiving peer could then request any
  transactions from the template that it was missing, either adding them
  to the local mempool or storing them in a cache.  This would allow
  peers with divergent mempool policies to share transactions despite
  their differences and provides an alternative to the previous proposal
  of using _weak blocks_ instead (see [Newsletter #299][news299 weak

>   blocks]).  Towns provided a [proof of concept implementation][towns

Reviewer: ```suggestion
  blocks]).  Towns provided a [proof-of-concept implementation][towns
```

Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
repo], and [BINANAs][binana repo]._

- [Bitcoin Core #32941][] p2p: TxOrphanage revamp cleanups

- [Bitcoin Core #31385][] package validation: relax the package-not-child-with-unconfirmed-parents rule

- [Bitcoin Core #31244][] descriptors: MuSig2

- [Bitcoin Core #30635][] rpc: add optional blockhash to waitfornewblock, unhide wait methods in help

- [Bitcoin Core #28944][] wallet, rpc: add anti-fee-sniping to `send` and `sendall`

- [Eclair #3133][] Add outgoing reputation

- [LND #10097][] Roasbeef/gossip-block-fix

- [LND #9625][] Add deletecanceledinvoice RPC call

- [Rust Bitcoin #4730][] p2p: Add formal `Alert` type

- [BLIPs #55][] Webhook Registration (LSPS5) (#55)

## Correction

In [last week's newsletter][news365 p2qrh], we incorrectly described the
updated version of [BIP360][], _pay to quantum-resistant hash_, as
"making exactly the change" that Tim Ruffing showed was secure in his
recent paper.  What BIP360 actually does is replaces the elliptical
curve commitment to a SHA256-based merkle root (plus a keypath

> alternative) with the merkle root directly.  Ruffing's paper showed that

Reviewer: In this newsletter, merkle is always lowercase: https://github.com/bitcoinops/bitcoinops.github.io/blob/master/STYLE.md#common-nouns

taproot, as currently used, is secure if a quantum-resistant signature
scheme were added to the [tapscript][topic tapscript] language and

> keypath spends were disabled.  BIP360 instead requires wallets upgrade

Reviewer: ```suggestion
keypath spends were disabled.  BIP360 instead requires that wallets upgrade
```
Or maybe "requires wallets to upgrade"

to a variant on taproot (albeit, a trivial variant), eliminates the
keypath mechanism from its variant, and describes the addition of a
quantum-resistant signature scheme to the scripting language used in its

> tapleaves.

Author: Taken, with two small edits.  Thanks!

[Bitcoin++ Insider][] has begun publishing reader-funded news about
technical Bitcoin topics.  Two of their free weekly newsletters, _Last
Week in Bitcoin_ and _This Week in Bitcoin Core_, may be especially
interesting to regular readers of the Optech newsletter.

## Releases and release candidates

_New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates._

- [Bitcoin Core 29.1rc1][] is a release candidate for a maintenance
  version of the predominant full node software.

- BTCPay Server 2.2.0 FIXME:harding

## Notable code and documentation changes

_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
repo], and [BINANAs][binana repo]._

> - [Bitcoin Core #32941][] completes the overhaul of `TxOrphanage` (see
>   [Newsletter #364][news364 orphan]) by enabling automatic trimming of the
>   orphanage whenever its limits are exceeded and by tracking only one
>   announcement per wtxid. The obsolete `maxorphantxs` option has been reinstated
>   but only emits a warning message. This PR solidifies opportunistic
>   one-parent-one-child (1p1c) [package relay][topic package relay].

Reviewer 2: ```suggestion
- [Bitcoin Core #32941][] completes the overhaul of `TxOrphanage` (see
  [Newsletter #364][news364 orphan]) by enabling automatic trimming of the
  orphanage whenever its limits are exceeded.
  It adds a warning for `maxorphantx` users fo inform them that it is obselete.
  This PR solidifies opportunistic
  one-parent-one-child (1p1c) [package relay][topic package relay].
```

- [Bitcoin Core #31385][] relaxes the
  `package-not-child-with-unconfirmed-parents` rule of the `submitpackage` RPC

>   to improve 1p1c [package relay][topic package relay] usage. Packages with
>   multiple unconfirmed parents can now be submitted alongside only one
>   unconfirmed parent.

Reviewer 2: Sorry, I know the PR is really confusing!
```suggestion
  to improve 1p1c [package relay][topic package relay] usage. Packages with
  no longer need to include the parents that are already in the node's mempool.
```

- [Bitcoin Core #31244][] implements the parsing of [MuSig2][topic musig]
  [descriptors][topic descriptors] as defined in [BIP390][], which is required
  for receiving and spending inputs from [Taproot][topic taproot] addresses with
  MuSig2 aggregate keys.

- [Bitcoin Core #30635][] adds an optional `current_tip` argument to the
  `waitfornewblock` RPC, to mitigate against race conditions by specifying the
  block hash of the current chain tip. This PR also begins displaying the

>   `waitfornewblock`, `waitforblock`, and `waitforblockheight` RPCs in the help

Author: I think @bitschmidty is just suggesting we emphasize that these are now public RPCs rather than hidden debug-only RPCs (which we usually only cover in special cases).

  If a peer sends a request before the previous one finishes, the additional
  message is quietly dropped. A new `gossip.filter-concurrency` setting (default
  5) is added to limit the number of concurrent workers across all peers. The PR
  also adds documentation explaining how all gossip rate limit configuration
  settings work.

- [LND #9625][] adds a `deletecanceledinvoice` RPC command (and its `lncli`
  equivalent) that allows users to remove canceled [BOLT11][] invoices (see
  [Newsletter #33][news33 canceled]) by providing their payment hash.

- [Rust Bitcoin #4730][] adds an `Alert` type wrapper for the [final alert][]
  message that notifies peers running a vulnerable version of Bitcoin Core
  (before 0.12.1) that their alert system is insecure. Satoshi introduced the
  alert system to notify users of significant network events, but it was
  [retired][] in version 0.12.1, except for the final alert message.

- [BLIPs #55][] adds [BLIP55][] to specify how mobile clients can register for
  webhooks via an endpoint to receive push notifications from an LSP. This
  protocol is useful for clients to get notified when receiving an [async

>   payment][topic async payment], and was recently implemented in LDK (See

Reviewer 3: ```suggestion
  payment][topic async payments], and was recently implemented in LDK (See
```

  export."
  a4link="https://bitcoincore.reviews/32489#l-100"

  q5="In `wallet_exported_watchonly.py`, why does the test call
  `wallet.keypoolrefill(100)` before checking spendability across the
  online/offline pair?"
  a5="The `keypoolrefill(100)` call forces the offline (spending) wallet
  to pre-derive 100 keys for its hardened descriptors, populating its
  cache. This cache is then included in the export, allowing the online
  watch-only wallet to generate those 100 addresses. It also ensures the
  offline wallet will recognize these addresses when it receives a PSBT
  to sign."
  a5link="https://bitcoincore.reviews/32489#l-122"
%}

## Optech recommends

[Bitcoin++ Insider][] has begun publishing reader-funded news about
technical Bitcoin topics.  Two of their free weekly newsletters, _Last
Week in Bitcoin_ and _This Week in Bitcoin Core_, may be especially
interesting to regular readers of the Optech newsletter.

## Releases and release candidates

_New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates._

> - [LND v0.19.3-beta.rc1][] is a maintenance release for this popular LND

Reviewer 4: ```suggestion
- [LND v0.19.3-beta.rc1][] is a maintenance release for this popular LN
```
